src/reader/slice_reader.rs

```diff
@@ -24,7 +24,7 @@ impl<'a> Reader<&'a [u8]> {
         #[cfg(feature = "encoding")]
         {
             let mut reader = Self::from_reader(s.as_bytes());
-            reader.encoding = EncodingRef::Explicit(UTF_8);
+            reader.parser.encoding = EncodingRef::Explicit(UTF_8);
```

Comment:
Encoding doesn't feel like it belongs on the parser.  It's primarily an aspect of reading but the parser can override the original default.  But as I mentioned on the other PR even the manual override isn't really something that encoding_rs nor encoding_rs_io will give you.

Reply:
Encoding is an interesting field. In can be part of configuration (when you create reader `from_str`) and it can be part of the parse state (when you create reader by other functions). Because I plan to use that in such manner:
```rust
/// Convert any synchronous reader to asynchronous one if inner reader support that
impl<R: AsyncBufRead + Unpin> From<Reader<R>> for Reader<TokioReader<R>> {
    fn from(reader: Reader<R>) -> Self {
        Self {
            reader: TokioReader(reader.reader),
            parser: reader.parser,
        }
    }
}

/// Convert any asynchronous reader to a synchronous one if inner reader support that
impl<R> From<Reader<TokioReader<R>>> for Reader<R> {
    fn from(reader: Reader<TokioReader<R>>) -> Self {
        Self {
            reader: reader.reader.0,
            parser: reader.parser,
        }
    }
}
```
it is convenient to leave it in the `Parser`. It will be possible to move it later if it is required